src/utils/RunnersUtils.js

```diff
@@ -32,7 +32,20 @@ const patchRunnerWithCurrentUserPermissions = (runner, userEmail, userId, permis
   };
 };
 
+const updateParentIdOnDelete = (runners, deletedRunnerId) => {
+  const parentId = runners.find((runner) => runner.id === deletedRunnerId)?.parentId;
+  const childrenIds = runners.filter((runner) => runner.parentId === deletedRunnerId)?.map((runner) => runner.id);
+  if (childrenIds.length > 0) {
+    runners.forEach((runner) => {
+      if (childrenIds.includes(runner.id)) {
+        runner.parentId = parentId ?? null;
+      }
+    });
+  }
```

Comment:
Maybe I'm missing something, but I think we could simplify this by doing only one `forEach` loop without the first `filter` + `map`. Something like:
```
const parentId = runners.find((runner) => runner.id === deletedRunnerId)?.parentId;
runners.forEach((runner) => {
  if (runner.parentId !== deletedRunnerId) return; // Skip datasets that must remain unchanged
  runner.parentId = parentId ?? null;
});
```

Reply:
Good idea, thanks ! :pray: